pulsar/producer_partition.go

```diff
@@ -292,6 +294,18 @@ func (p *partitionProducer) reconnectToBroker() {
 }
 
 func (p *partitionProducer) runEventsLoop() {
+	go func() {
+		for {
+			select {
+			case <-p.doneCh:
+				return
+			case <-p.connectClosedCh:
```

Comment:
In the event of a broker being down we will not be able to close Pulsar because it will be stuck inside the reconnectToBroker method.

Reply:
This is already taken care by this check within `reconnectToBroker` -  https://github.com/hriday-ns/pulsar-client-go/blob/51854f7d8e398d5580fb7fceb252de69c62ad59a/pulsar/producer_partition.go#L274-L277
I verified this and this check would make sure we exist out of the forever loop when the producer is closed